### ISSUE TITLE
Rule await-async-query "false positive" with other findBy*s

### ISSUE THREAD

**Issue by JanHuege** [testing-library/eslint-plugin-testing-library#498]
### Plugin version

v4.12.4

### ESLint version

7.2.0

### Node.js version

14.17.4

### npm/yarn version

yarn 2.4.2

### Operating system

appears on mac, windows, does not matter

### Bug description

Analysis for hierarchies with Call- and Memberexpressions are not correctly detected if one member identifier contains "findBy" pattern and does not return a promise.

Example: Using mongoose to test/assert/expect if something was written to the database.

### Steps to reproduce

Examples to reproduce:

1. 
```
const findByTest = () => ({ a: () => Promise.resolve("TEST")})

const foo = async () => {
    const result = await findByTest().a();
}
```

2.
`const a = findByTest(); `

### Error output/screenshots

  173:51  error  promise returned from `findById` query must be handled  testing-library/await-async-query

✖ 1 problem (1 error, 0 warnings)

### ESLint configuration

```        
{
  "files": [
      "**/__tests__/**/*.*"
  ],
  "rules": {
      "no-console": "off"
  },
  "extends": [
      "plugin:testing-library/react",
      "plugin:jest-dom/recommended"
  ]
}
```

### Rule(s) affected

testing-library/await-async-query

### Anything else?

Do not know right now if I will submit a fix. Might change in the future.

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Hey @JanHuege! This is related to the Aggressive Reporting feature. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior? In that section, you have the info for restricting the plugin so it only reports built-in queries.

**Comment by JanHuege** [comment]
Hi @Belco90. Sorry for the late response. I looked into it and it worked. Thanks. Looks like I did not scroll enough in that readme. Maybe a ToC at the top would be nice :)

**Comment by MichaelDeBoey** [comment]
@JanHuege GitHub already provides a ToC out of the box
https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files

**Comment by JanHuege** [comment]
Got another issue/topic related to this one not handled by turning off aggressive reporting. Inside an awaited promise-chain there are also false-positives if using testcafe for which testinglib selectors are also available:

```
    await t
        .pressKey('enter')
        .pressKey('esc')
        .click(screen.findByText('test'))
        .click(saveButton);
```

**Comment by Belco90** [comment]
Hi again. I'm afraid testcafe isn't fully supported by this plugin: only DOM, Angular, React, and Vue are.
